File: CPP/Subarray/Maximum Unique Subarray Sum After Deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        set<int>st;
        for(auto num:nums){
            if(num>0){
                st.insert(num);
            }
        }
        if(st.size()==0)return *max_element(nums.begin(),nums.end());
        return accumulate(st.begin(),st.end(),0);
    }
};

File: Java/HashSet/Maximum Unique Subarray Sum After Deletion.java
class Solution {
    public int maxSum(int[] nums) {
        HashSet<Integer> st=new HashSet<>();
        for(int num:nums){
            if(num>0){
                st.add(num);
            }
        }
        if (st.size()==0){
            int maxi=Integer.MIN_VALUE;
            for(int num:nums){
                if(num>maxi)maxi=num;
            }
            return maxi;
        }
        int max_sum=0;
        for(Integer i:st)max_sum+=i;
        return max_sum;
    }
}
